### DISCUSSION TITLE
Setting graphical parameters with `par()` does not work?

### DISCUSSION THREAD

**Discussion by cbrnr** [quarto-dev/quarto-cli#3444] [Q&A] [answered]
I was trying to create a plot after setting a parameter with `par()`, but the changes do not seem to persist.

In the following example, the second plot should be red, but it's still black. Is this a bug?

````
---
title: "test"
format: html
---

First a plot with default parameters:

```{r}
plot(1:10)
```

Now let's change the color:

```{r}
par(col="red")
```

And we plot again:

```{r}
plot(1:10)
```
````

**Comment by cscheid** [comment, marked as answer]
This is a [knitr feature](https://bookdown.org/yihui/rmarkdown-cookbook/global-device.html). You need 

```r
knitr::opts_knit$set(global.device = TRUE)
```

**Reply by cbrnr** [reply]
Thanks @cscheid!

**Comment by loreabad6** [comment]
Hi! I tested this solution either by adding a chunk with `knitr::opts_knit$set(global.device = TRUE)` or by adding it to the YAML as: 

```
knitr:
  opts_knit:
    global.device: true
```

In both cases I have a weird behavior when rendering, where the initial plot is rendered after every code chunk. I can't attach the html file but here is a screenshot: 
![image](https://github.com/user-attachments/assets/d5fa9fe5-c2b3-4b15-a12b-aef259d2a4f3)

I can't test this right now on the latest Quarto version, testing now on 1.4.555 and 1.3.xx.

I assume this is a bug, but I wait for confirmation if someone can test in 1.5 to report it. Thanks!

**Reply by mcanouil** [reply]
Could you open a new discussion and could you share a **small self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? The goal is to make it as easy as possible for us to recreate your problem so that we can fix it: please help us help you! Thanks.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by loreabad6** [reply]
thanks @mcanouil

Should I go for a discussion or directly for an issue?

**Reply by mcanouil** [reply]
It does not really matter if you provide all the information requested in the template, especially in the issue template.

If you are not completely sure, a discussion might be better. (we can still open an issue from the discussion if needed)

**Reply by cderv** [reply]
Issue opened at 
- https://github.com/quarto-dev/quarto-cli/issues/10302
